ee/src/main/java/org/jboss/as/ee/component/deployers/ModuleJndiBindingProcessor.java

```diff
@@ -167,6 +172,10 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
                 if(config.isInvalid()) {
                     continue;
                 }
+                ClassInfo classInfo = index.getClassByName(DotName.createSimple(config.getClassName()));
+                if (!isConcreteClass(classInfo) && !hasConcreteSubclass(index, classInfo)) {
+                    continue;
+                }
                 final Set<BindingConfiguration> classLevelBindings = new HashSet<>(config.getBindingConfigurations());
```

Comment:
I recommend moving L175-178 below this line and wrapping it in an  'if (!classLevelBindings.isEmpty())' check. The l175-178 scan is not cheap so it should be avoided whenever possible.